wagtail/admin/tests/test_userbar.py

```diff
@@ -495,6 +496,23 @@ def construct_wagtail_userbar(request, items):
 
             self.assertTrue(kwargs.get("called"))
 
+    @override_settings(
+        WAGTAIL_I18N_ENABLED=True,
+        WAGTAIL_CONTENT_LANGUAGES=[("en", "English"), ("fr", "French")],
+        WAGTAILADMIN_PERMITTED_LANGUAGES=[("en", "English")],
+        LANGUAGE_CODE="en",
+    )
+    def test_userbar_rendered_in_admin_permitted_language_if_user_has_no_language(self):
+        french = Locale.objects.create(language_code="fr")
+        self.homepage.copy_for_translation(french)
+        french_page = self.page.copy_for_translation(french)
+
+        with translation.override("fr"):
+            response = french_page.serve(self.request)
+            response.render()
+
+        self.assertContains(response, "Go to Wagtail admin")
```

Comment:
thought: it would be good to test the other 2 test cases:
1. the user's preferred language
2. the active laguage is in the available admin languages

Reply:
Hah yes, that was indeed needed!